front/app/components/EventAttendanceButton/index.tsx

```diff
@@ -92,11 +91,17 @@ const EventAttendanceButton = ({ event }: EventAttendanceButtonProps) => {
     }
 
     if (disabled_reason && isFixableByAuthentication(disabled_reason)) {
-      const context = {
-        type: 'phase',
-        action: 'attending_event',
-        id: currentPhase?.id,
-      } as const;
+      // If no current phase, we still allow event registration through the default flow
+      const context = currentPhase
+        ? ({
+            type: 'phase',
+            action: 'attending_event',
+            id: currentPhase?.id,
+          } as const)
+        : ({
```

Comment:
I don't really like that we use the global flow for this. This is an action in a project context, which depends on e.g. the publication status and visibility status of the project, not on settings related to the global context...

In general, if events are attributes of projects and not of phases, maybe we should also not treat `attending_event` as an action in a phase context.

But I assume it would be a massive amount of work to create a 'project' context type, and move the attending event logic in there?

Reply:
Yeah, that's the issue here. There isn't a project context that can be configured via permissions, they're only available on phases, so the fallback here is that we fall back to the global context as that is the default registration flow if I understand correctly. This way we get to have more specific flows in different phases, but still fallback to 'normal' registration when there is no phase.

It is certainly reliant on the project being there in this instance the code that sets this context will never be triggered unless the project is present.

Reply:
Have added a bit more information to the comment, to explain why